### DISCUSSION TITLE
Question about building locally

### DISCUSSION THREAD

**Discussion by tntdc585** [eclipse-embed-cdt/eclipse-plugins#595] [Debug Plug-ins Q&A]
When the eclipse-plugins plug-in is built locally using “mvn clean verify”, why does the build fail when there is only one change in the code, even if only one line of comments is added?

**Comment by jonahgraham** [comment]
How does it fail? My guess without an error message is that if the sources change since the last release then the version number needs to be bumped.

**Reply by tntdc585** [reply]
![QQ截图20231103164502](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/70255145/b86f93fd-730b-4000-aca5-8e18666ee89a)
![QQ截图20231103164510](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/70255145/320df494-5a55-493c-ba07-16293a789b2f)
Can I ask how to change the version number? If you modify the code, even if you just add a comment, you will get an error like this.

**Reply by jonahgraham** [reply]
Those errors are not related to my guess. I can't tell what has gone wrong here, perhaps there is some context above what you have shared. Can you attach a log of the full build (not screenshots of text please) or alternatively compare to the build machine to see what is different about your setup.

I just ran a build on the CI to make sure that the build isn't suffering from some bitrot, you can see the full console of the last successful build here: https://ci.eclipse.org/embed-cdt/job/build-plug-ins/job/develop/lastSuccessfulBuild/consoleFull

**Reply by jonahgraham** [reply]
PS your errors make it looks like the classpath has not been calculated successfully. It would be great if you do figure out what went wrong if you could add an answer here with what your solution was.

PPS Make sure you are using Java 11 and mvn 3.9.x to match the build machine.

**Reply by tntdc585** [reply]
![QQ截图20231106101338](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/70255145/5244ac4c-b696-4bbd-8d66-da24c52335a6)
[1.txt](https://github.com/eclipse-embed-cdt/eclipse-plugins/files/13261815/1.txt)
I just deleted the blank lines between lines 121 and 122, and the build failed, and then this error was reported. It is really strange.

**Reply by jonahgraham** [reply]
Why are there red underlines at lines 122 and 123? That indicates your dev environment is not setup correctly. If you make edits in the Eclipse editor with a broken setup the auto-cleanups may do things like remove imports. The best indication I can see is that `protected TabDebugger...` is not actually on line 120 in the original:

https://github.com/eclipse-embed-cdt/eclipse-plugins/blob/8eab829a01a4d47aae4a73c4d20b4233792484fd/plugins/org.eclipse.embedcdt.debug.gdbjtag.openocd.ui/src/org/eclipse/embedcdt/debug/gdbjtag/openocd/ui/TabDebugger.java#L122

That indicates two additional lines have been removed above line 120, perhaps a couple of key "import" lines?

Can you do a `git diff` and show all the changes, or commit the changes to your fork and I can try to build it from there.

**Comment by TommyMurphyTM1234** [comment]
> ![QQ截图20231106101338](https://user-images.githubusercontent.com/70255145/280576118-5244ac4c-b696-4bbd-8d66-da24c52335a6.png) [1.txt](https://github.com/eclipse-embed-cdt/eclipse-plugins/files/13261815/1.txt) I just deleted the blank lines between lines 121 and 122, and the build failed, and then this error was reported. It is really strange.

What happens if you try to build without any changes at all?

**Reply by tntdc585** [reply]
It will build success

**Comment by TommyMurphyTM1234** [comment]
> It will build success

Then you probably need to post a full build log as @jonahgraham previously suggested above:

* https://github.com/eclipse-embed-cdt/eclipse-plugins/discussions/595#discussioncomment-7470156

Without more detailed info it's impossible for anybody here to identify what the issue might be.

**Reply by jonahgraham** [reply]
The full log was included in the file 1.txt in this comment: https://github.com/eclipse-embed-cdt/eclipse-plugins/discussions/595#discussioncomment-7482670

**Comment by TommyMurphyTM1234** [comment]
> The full log was included in the file 1.txt in this comment: [#595 (reply in thread)](https://github.com/eclipse-embed-cdt/eclipse-plugins/discussions/595#discussioncomment-7482670)

Ah, sorry, didn't notice the attachment.

**Comment by tntdc585** [comment]
I solved the problem, thank you all. I imported this project using eclipse. As long as I modify the code (just add comments or delete blank lines), other things will be imported and modified, causing the class path calculation to fail. It does seem a bit strange. I used vscode to manage this project and there was no such problem.

**Reply by jonahgraham** [reply]
Very strange indeed - especially as VSCode's Java support is provided by Eclipse JDT (i.e. the engine is the same for both)
